keyrings/alt/file_base.py

```diff
@@ -71,35 +108,44 @@ def get_password(self, service, username):
         try:
             password_base64 = config.get(service, username).encode()
             # decode with base64
-            password_encrypted = base64.decodestring(password_base64)
-            # decrypted the password
-            password = self.decrypt(password_encrypted).decode('utf-8')
+            password_encrypted = decodebytes(password_base64)
+            # decrypt the password with associated data
+            try:
+                password = self.decrypt(password_encrypted, assoc).decode('utf-8')
+            except ValueError:
+                # decrypt the password without associated data
+                password = self.decrypt(password_encrypted).decode('utf-8')
         except (configparser.NoOptionError, configparser.NoSectionError):
             password = None
         return password
 
     def set_password(self, service, username, password):
         """Write the password in the file.
         """
-        service = escape_for_ini(service)
-        username = escape_for_ini(username)
-
+        assoc = (escape_for_ini(service) + '\0' +
```

Comment:
Yes, the divergence is due to avoidance of repeated escapes of the same value in get_password(), that would happen, if consolidated (probably a case of premature optimization).

We reuse the escaped values in get_password, while preserving the unescaped values in set_password (as escaping is delayed to _write_config_values in this code path).